### DISCUSSION TITLE
Can't send PING

### DISCUSSION THREAD

**Discussion by a-moshchenko** [miguelgrinberg/python-socketio#902] [Q&A] [answered]
**Describe the bug**
HI! I have connection problems

**Expected behavior**
If not started  PING PONG, I have error "packet queue is empty, aborting" and connection dicconnect

**Logs**
python-engineio==3.13.2
python-socketio==4.6.0

Attempting WebSocket connection to wss://agnj3.866rt7hh.com/socket.io/?gid=c46cd3394b0b0c28&token=9689657a-615e-4842-ba93-a2a76fa53945&id=340u02z41nkuadd3coqpym4p&rid=2&EIO=3&transport=websocket&EIO=3
WebSocket connection accepted with {'sid': 'bper6be4-pbdcyM_AdSG', 'upgrades': [], 'pingInterval': 8000, 'pingTimeout': 12000}
Engine.IO connection established
Sending packet PING data None       `<------- HERE`
Received packet MESSAGE data 0
Namespace / is connected
Received packet MESSAGE data 2["init",{"pnid":"","sm":2}]
Received event "init" [/]
Emitting event "init" [/]
Sending packet MESSAGE data 2["init",{"gid":"c46cd3394b0b0c28","token":"9689657a-615e-4842-ba93-a2a76fa53945","id":"340u02z41nkuadd3coqpym4p","rid":"2","v":2}]
Emitting event "subscribe" [/]
Sending packet MESSAGE data 2["subscribe",["odds",[{"id":"c3","rev":"","sorting":"t","condition":{"sporttype":1,"marketid":"L","bettype":[1,3,5,7,8,15,301,303]}}]]]
connected

python-engineio @ git+https://github.com/miguelgrinberg/python-engineio@a9fb317c29d93b98a729df800f124b6b923fc82c
python-socketio @ git+https://github.com/miguelgrinberg/python-socketio@b4b55e1fb5e9d837b3301b5b873d9a7fe2a12023

Attempting WebSocket connection to wss://agnj3.866rt7hh.com/socket.io/?gid=b3b0167a4570380d&token=32fe8a7c-faae-4285-a9c2-ab7feed1078f&id=hwec5xplfgot5se0wt1cce02&rid=2&EIO=3&transport=websocket&EIO=4
WebSocket connection accepted with {'sid': 'Jm6_H0xOggPL2MBpAc3L', 'upgrades': [], 'pingInterval': 8000, 'pingTimeout': 12000}
Engine.IO connection established
Sending packet MESSAGE data 0{}         `<----  AND HERE`   
Received packet MESSAGE data 0
Namespace / is connected
Received packet MESSAGE data 2["init",{"pnid":"","sm":2}]
Received event "init" [/]
Emitting event "init" [/]
Sending packet MESSAGE data 2["init",{"gid":"b3b0167a4570380d","token":"32fe8a7c-faae-4285-a9c2-ab7feed1078f","id":"hwec5xplfgot5se0wt1cce02","rid":"2","v":2}]
Emitting event "subscribe" [/]
Sending packet MESSAGE data 2["subscribe",["odds",[{"id":"c3","rev":"","sorting":"t","condition":{"sporttype":1,"marketid":"L","bettype":[1,3,5,7,8,15,301,303]}}]]]
connected

*i whant to use latest version*

**Comment by miguelgrinberg** [comment, marked as answer]
Please don't write issues when you need assistance troubleshooting your application. The issues board is for tracking bugs in this project. The discussion board is to troubleshoot your own application's problems.

The problem that you have is not clear from your description. I don't see anything in the logs that you included that indicates a problem. The first log is for an old version of this package and the Socket.IO protocol, where the client issued PING packets and the server responded with PONG. In the latest revision of the Socket.IO protocol, the server sends the PING and the client responds with PONG. Your client must implement the same version of the Socket.IO protocol as the server for things to work.
